lib/token-autocomplete.ts

```diff
@@ -492,27 +492,46 @@ class TokenAutocomplete {
                         highlightedSuggestion = parent.autocomplete.suggestions.firstChild;
                     }
 
-                    me.handleInput(highlightedSuggestion);
+                    this.handleInput(highlightedSuggestion);
                 } else if (parent.getCurrentInput() === '' && event.key == parent.KEY_BACKSPACE) {
                     event.preventDefault();
-                    me.removeLastToken();
+                    this.removeLastToken();
                 }
                 if ((event.key == parent.KEY_DOWN || event.key == parent.KEY_UP) && parent.autocomplete.suggestions.childNodes.length > 0) {
                     event.preventDefault();
                 }
             });
 
-            parent.textInput.addEventListener('keyup', function (event) {
-                me.updateHasValue();
+            parent.textInput.addEventListener('keyup', () => this.updateHasValue());
+
+            parent.textInput.addEventListener('focusout', (event: FocusEvent) => {
+                // Using setTimeout here seems hacky on first sight but ensures proper order of events / handling.
+                // We first want to handle a click on a suggestion (when one is made) before hiding the suggestions on focusout of the input.
+                // Not doing so could mean the suggestion is hidden before the click is handled und thus resulting in not being selected.
+                // This depends on the order in which a browser handles different events and when it sets the active pseudo-selector on clicked events (Firefox for example)
+                setTimeout(() => {
+                    if (parent.autocomplete.areSuggestionsActive()) {
+                        return;
+                    }
+                    const input = this.parent.getCurrentInput();
+                    if (input != '' && !this.handleInputAsValue(input)) {
+                        this.container.dispatchEvent(new CustomEvent('input-ignored', {
```

Comment:
Dokumentation vllt bitte noch in die [README.md](https://github.com/scireum/token-autocomplete/blob/main/README.md) aufnehmen und die Beschreibung der anderen Events ggf. noch anpassen an das neue Verhalten.